cocotb/share/lib/vhpi/VhpiCbHdl.cpp

```diff
@@ -268,7 +269,9 @@ int VhpiSignalObjHdl::initialise(std::string &name, std::string &fq_name) {
         case vhpiStrVal: {
             m_indexable = true;
             m_num_elems = vhpi_get(vhpiSizeP, handle);
-            m_value.bufSize = (m_num_elems)*sizeof(vhpiCharT) + 1;
+            m_value.bufSize = (m_num_elems)*static_cast<decltype(m_value.bufSize)>(
```

Comment:
We should probably introduce a typedef for this decltype, and use it for all of the size of casts below. Will come back to this with a suitable `using bufsize_t = decltype(...)` later.